drivers/include/tcs37727.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2015 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_tcs37727 TCS37727 Light-To-Digital Converter
+ *
+ * @ingroup     drivers
+ * @brief       Driver for the AMS TCS37727 Color Light-To-Digital Converter
+ *
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the TCS37727 sensor driver.
+ *
+ * @author      Felix Siebel <f.siebel@phytec.de>
+ */
+
+#ifndef TCS37727_H
+#define TCS37727_H
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#ifndef TCS37727_I2C_ADDRESS
+#define TCS37727_I2C_ADDRESS           0x29 /**< Default Device Address */
+#endif
+
+/**
+ * @brief Device descriptor for TCS37727 sensors.
+ */
+typedef struct {
+    i2c_t i2c;              /**< I2C device the sensor is connected to */
+    uint8_t addr;           /**< the sensor's slave address on the I2C bus */
+    bool initialized;       /**< sensor status, true if sensor is initialized */
+} tcs37727_t;
+
+/**
+ * @brief TCS37727 sensor test.
+ * This function looks for Manufacturer ID of the TCS37727 sensor.
+ *
+ * @param[in]  dev          device descriptor of sensor
+ *
+ * @return                  0 on success
+ * @return                  -1 on error
+ */
+int tcs37727_test(tcs37727_t *dev);
+
+/**
+ * @brief Initialise the TCS37727 sensor driver.
+ * Settings: Gain 4x, Proximity Detection off
+ *
+ * @param[out] dev          device descriptor of sensor to initialize
+ * @param[in]  i2c          I2C bus the sensor is connected to
+ * @param[in]  address      sensor's I2C slave address
+ *
+ * @return                  0 on success
+ * @return                  -1 if initialization of I2C bus failed
+ * @return                  -2 if sensor test failed
+ * @return                  -3 if sensor configuration failed
+ */
+int tcs37727_init(tcs37727_t *dev, i2c_t i2c, uint8_t address);
+
+/**
+ * @brief Read raw sensor's data.
+ *
+ * @param[in]  dev          device descriptor of sensor
+ * @param[out] rawred       raw Red ADC data
+ * @param[out] rawgreen     raw Green ADC data
+ * @param[out] rawblue      raw Blue ADC data
+ * @param[out] rawclear     raw Clear ADC data
+ *
+ * @return                  0 on success
+ * @return                  -1 on error
+ */
+int tcs37727_read(tcs37727_t *dev, uint16_t *rawred, uint16_t *rawgreen,
```

Comment:
I would like to see this driver in a similar style as for the [isl29125](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/isl29125.h). This would mean to use the `color_rgb_t` as output type. If you think you also need a higher resolution you could also introduce an own type in addition (just compare the isl29125 implementation)